forte/data/data_store.py

```diff
@@ -939,6 +943,49 @@ def _parse_onto_file(self):
             return
         raise NotImplementedError
 
+    @staticmethod
+    def get_class(type_name):
+        r"""Returns the ontology class based on class name. Only a class that exists in current
+        DataStore will be returned.
+
+        Args:
+            type_name (str): Fully qualified name of the class.
+        Returns:
+            The target class.
+
+        Raises:
+            ValueError: If class is not found based on :attr:`class_name` in DataStore.
+        """
+        if type_name not in DataStore._type_attributes:
+            raise ValueError(f"Class not found in DataStore: {type_name}")
+        return get_class(type_name)
+
+    @staticmethod
+    def get_subtypes(type_name):
+        r"""Returns a complete list of subclasses based on class name.
+
+        Args:
+            type_name (str): Fully qualified name of the class.
+        Returns:
+            A list of subtypes.
+        """
+        if DataStore.enable_cache:
+            try:
+                return DataStore._get_subtype_cache[type_name]
```

Comment:
So I can see the following scenario to be a problem: 

- We call the subtype of a particular class A.
- We then add a new entry with class A_1, which is a subclass of A, but not known before
- We call this function because the cache is enabled/valid, this will return the old (wrong set).

The solution would be:
1. We clear the subtype_cache when adding unknown class A
2. When adding the new class A, we update subtype_cache

Also, make sure you add these to the test cases.

Note that only disabling `enable_cahce` is not enough, because if the data (subtype_cache) is there, and something else triggers the cache, this will still cause a problem.